src/bioregistry/schema/struct.py

```diff
@@ -669,6 +669,78 @@ def get_pattern_re(self):
             return None
         return re.compile(pattern)
 
+    def get_pattern_with_banana(self, strict: bool = True) -> Optional[str]:
+        """Get the pattern for the prefix including a banana if available.
+
+        .. warning::
+
+            This function is meant to mediate backwards compatibility with legacy
+            MIRIAM/Identifiers.org standards. New projects should **not** use redundant
+            prefixes in their local unique identifiers.
+
+        :param strict: If True (default), and a banana exists for the prefix,
+            the banana is required in the pattern. If False, the pattern
+            will match the banana if present but will also match the identifier
+            without the banana.
+        :returns: A pattern for the prefix if available
+
+        >>> import bioregistry as br
+        >>> resource = br.get_resource("chebi")
+
+        Strict match requires banana
+        >>> resource.get_pattern_with_banana()
+        '^CHEBI:\\d+$'
+        >>> resource.get_pattern_with_banana(strict=False)
+         '^(CHEBI:)?\\d+$'
+        """
+        pattern = self.get_pattern()
+        if pattern is None:
+            return None
+        banana = self.get_banana()
+        if not banana:
+            return pattern
+
+        banana_peel = self.get_banana_peel()
+        prepattern = f"{banana}{banana_peel}"
+        if not strict:
+            prepattern = f"({prepattern})?"
+        return "^" + prepattern + pattern.lstrip("^")
+
+    def get_pattern_re_with_banana(self, strict: bool = True):
```

Comment:
I'm curious, is it problematic to add the `re.Pattern` annotation for the return value?

Reply:
It’s changed so much since py37 that idk a way to do it without writing a compatibility layer